tests/test_nodes.py

```diff
@@ -89,6 +90,59 @@ def test_insert_issue_in_a_more_complex_situation():
     assert str(document) == "<root><div1><div2/> </div1></root>"
 
 
+def test_an_issue():
+    def node_ids():
+        return {
+            "root": id(root),
+            "foo": id(foo),
+            "div1": id(div1),
+            "div2": id(div2),
+            "text": id(text),
+        }
+
+    document = Document("<root><foo><div1><div2/>text</div1></foo></root>")
+
+    root = document.root
+    foo = root.first_child
+    div1 = foo.first_child
+    div2 = div1.first_child
+    text = div1.last_child
+
+    original_ids = node_ids()
+
+    div1.detach()
+    foo = root.first_child
+    div2 = div1.first_child
+    text = div1.last_child
+    assert node_ids() == original_ids
+
+    foo.detach()
+    assert node_ids() == original_ids
+
+    root.insert_child(0, div1)
+    div1 = root.first_child
+    div2 = div1.first_child
+    text = div1.last_child
+    assert node_ids() == original_ids
+
+    # other_doc = Document(str(document))
```

Comment:
Are the commented blocks still needed or maybe even meant to be un-commented? Apparently there's no harm in running them so what's the point in having them commented out?

Reply:
as above.